backend/src/main/java/com/capstone/moneytree/facade/YahooFinanceFacade.java

```diff
@@ -0,0 +1,22 @@
+package com.capstone.moneytree.facade;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Component
+public class YahooFinanceFacade {
+
+    private final WebClient webClient;
+
+    @Autowired
+    public YahooFinanceFacade(WebClient.Builder webClientBuilder) {
+        this.webClient = webClientBuilder.baseUrl("https://query1.finance.yahoo.com").build();
```

Comment:
Refactor the URL string maybe in the properties or on top as a private final variable.

Reply:
good idea

Reply:
Fair enough.